kernel/kernel-impl/src/main/java/org/sakaiproject/time/impl/UserTimeServiceImpl.java

```diff
@@ -169,10 +173,9 @@ public String dateTimeFormat(Instant date, FormatStyle dateStyle, FormatStyle ti
         if (date == null) return "";
         if (dateStyle == null) { dateStyle = FormatStyle.MEDIUM; }
         if (timeStyle == null) { timeStyle = FormatStyle.SHORT; }
-        Locale locale = preferencesService.getLocale(sessionManager.getCurrentSessionUserId());
```

Comment:
What happens when there is no site locale set? And the user has a preferred locale?

Reply:
Then, dates are shown on user preferred locale language.

Reply:
Great. Where does that logic happen?

Reply:
On the next line: .withLocale(resourceLoader.getLocale());

https://github.com/sakaiproject/sakai/pull/10187/files#diff-c381b6340f804a20160ea067d2a476a4eaa3b7a1b369347fd9cf8b8d970a99baR178